go/vt/topo/vschema.go

```diff
@@ -20,34 +20,58 @@ import (
 	"context"
 	"path"
 
-	"google.golang.org/protobuf/proto"
-
 	"vitess.io/vitess/go/vt/log"
 	"vitess.io/vitess/go/vt/vterrors"
 
 	vschemapb "vitess.io/vitess/go/vt/proto/vschema"
 )
 
-// SaveVSchema saves a Vschema. A valid Vschema should be passed in. It does not verify its correctness.
-// If the VSchema is empty, just remove it.
-func (ts *Server) SaveVSchema(ctx context.Context, keyspace string, vschema *vschemapb.Keyspace) error {
+// KeyspaceVSchemaInfo wraps a vschemapb.Keyspace and is a meta
+// struct that contains metadata to give the data more context
+// and convenience. This is the main way we interact with a
+// keyspace's vschema.
+type KeyspaceVSchemaInfo struct {
+	Name string
+	*vschemapb.Keyspace
+	version Version
+}
```

Comment:
This struct is a duplicate of an existing one in this same package. There exists `KeyspaceInfo` in keyspace.go. The struct definition is as follows - 
```go
// KeyspaceInfo is a meta struct that contains metadata to give the
// data more context and convenience. This is the main way we interact
// with a keyspace.
type KeyspaceInfo struct {
	keyspace string
	version  Version
	*topodatapb.Keyspace
}
```

It pretty much has the same information just with a different nomenclature. I think we should remove this and use that struct directly.

Reply:
`KeyspaceInfo` is for the topo Keyspace record: https://github.com/vitessio/vitess/blob/eaaa2063ef870391bf17dbf3599978c4eba20868/proto/topodata.proto#L259-L301

`KeyspaceVSchemaInfo` is for the keyspace's vschema: https://github.com/vitessio/vitess/blob/eaaa2063ef870391bf17dbf3599978c4eba20868/proto/vschema.proto#L40-L60

Definitely not the same thing 🙂